src/gsi/read_obs.F90

```diff
@@ -487,6 +490,15 @@ subroutine read_obs_check (lexist,filename,jsatid,dtype,minuse,nread)
             endif
             nread = nread + 1
          end do loop_hdob
+       else if(trim(filename) == 'gnssrwndbufr')then
+         lexist = .false.
+         gnssrwndloop: do while(ireadmg(lnbufr,subset,idate2) >= 0)
+           if (trim(dtype)=='gnssrspd') then
+               lexist = .true.
+               exit gnssrwndloop
+           endif
+         end do gnssrwndloop
+
```

Comment:
Seems we should have the line `nread = nread + 1` here, right?

Reply:
We didn't seem to need this line, but I added it.